include/rnp/rnp.h

```diff
@@ -2252,12 +2258,32 @@ RNP_API rnp_result_t rnp_op_verify_detached_create(rnp_op_verify_t *op,
                                                    rnp_input_t      input,
                                                    rnp_input_t      signature);
 
+/**
+ * @brief Set additional flags which control data verification/decryption process.
+ *
+ * @param op pointer to opaque verification context.
+ * @param flags verification flags. ORed combination of RNP_VERIFY_* values.
+ *              Following flags are supported:
+ *              RNP_VERIFY_IGNORE_SIGS_ON_DECRYPT - ignore invalid signatures for the encrypted
+ *                and signed data. If this flag is set then rnp_op_verify_execute() call will
+ *                succeed and output data even if all signatures are invalid or issued by the
+ *                unknown key(s).
+ *              RNP_VERIFY_REQUIRE_ALL_SIGS - require that all signatures (if any) must be
+ *                valid for successful run of rnp_op_verify_execute().
+ * @return RNP_SUCCESS or error code if failed
+ */
+RNP_API rnp_result_t rnp_op_verify_set_flags(rnp_op_verify_t op, uint32_t flags);
+
 /** @brief Execute previously initialized verification operation.
  *  @param op opaque verification context. Must be successfully initialized.
- *  @return RNP_SUCCESS if data was processed successfully and all signatures are valid.
- *          Otherwise error code is returned. After rnp_op_verify_execute()
- *          rnp_op_verify_get_* functions may be used to query information about the
- *          signature(s).
+ *  @return RNP_SUCCESS if data was processed successfully and output may be used. By default
+ *          this means at least one valid signature for the signed data, or successfully
+ *          decrypted data if no signatures are present.
```

Comment:
> or successfully  decrypted data if no signatures are present.

This does not look ok to me, if input data does not contain signatures, then RNP_ERROR_NO_SIGNATURES_FOUND must be returned, isn't it?

Reply:
@antonsviridenko When we start to process data, we do not know whether it is expected to be signed, as usually signatures are placed inside of the encrypted layer. User may easily check whether data was signed at all via `rnp_verify_get_signature_count()` and corresponding calls.

Reply:
Yeah, but purpose of this functions is to verify signatures, no signatures found => unsuccessful result should be returned.

Btw I've just found that error code `RNP_ERROR_NO_SIGNATURES_FOUND` is not used anywhere in verification code.

Reply:
We may add some flag to be set via `rnp_op_verify_set_flags()`, but do we really need that if user may check the case via the single `rnp_op_verify_get_signature_count()` call?